src/app/components/known-targets-histogram/known-targets-histogram.component.ts

```diff
@@ -0,0 +1,96 @@
+import { Component } from '@angular/core';
+import { ElementRef, SimpleChange, OnChanges, ViewChild,
+         AfterViewInit, ViewEncapsulation, Input, OnInit } from '@angular/core';
+import { Settings } from '../../models/settings';
+import { KnownTargets } from '../../models/known-targets';
+import { Observable } from 'rxjs';
+import { Store } from '@ngrx/store';
+import { ManipulateDataService } from '../../services/manipulate-data.service';
+import * as fromRoot from '../../reducers';
+
+import * as d3 from 'd3';
+
+@Component({
+  selector: 'app-known-targets-hist',
+  encapsulation: ViewEncapsulation.Native,
+  styleUrls: ['./known-targets-histogram.component.scss'],
+  templateUrl: './known-targets-histogram.component.html'
+})
+
+export class KnownTargetsHistogramComponent implements  AfterViewInit, OnInit {
+    graphData: Array<Number>;
+    data: Array<number>;
+    width: string;
+    height: string;
+    @ViewChild('container') element: ElementRef;
+    @Input() settings: Settings;
+    private el: HTMLElement;
+    private margin =  {top: 16, right: 48, bottom: 16, left: 8};
+    private padding = {top: 16, right: 24, bottom: 16, left: 24};
+    private divs: any;
+    knownTargets: KnownTargets;
+    knownTargetsFetched$: Observable<boolean>;
+    dataset: Array<{name: string, value: number}>;
+    datasetNames: Array<string>;
+    datasetValues: Array<number>;
+
+    constructor(private store: Store<fromRoot.State>,
+    private manipulateDataService: ManipulateDataService) {
+      this.knownTargetsFetched$ = this.store.let(fromRoot.getKnownTargetsFetched);
+      this.dataset = [];
+      this.datasetNames = [];
+      this.datasetValues = [];
+    }
+
+    ngOnInit() {
+      this.knownTargetsFetched$
+        .subscribe(ev => {if (ev) {
```

Comment:
OK, I understand the server gives a `200 OK` even when there are errors, so the error will not be captured by the Observable. It's still missing the Observable.error function in case there is a network or server error.

Regarding the API for errors, the current code is also not capturing all errors, for example:
```
{
  "status": "ERROR",
  "result": "Cannot parse config: String: 1: Key 'pwids' may not be followed by token: end of file (if you intended 'pwids' to be part of a value, instead of a key, try adding double quotes around the whole value)"
}
```
The result is not empty, but there's still an error.